src/shared/infrastructure/http/__tests__/ApiClient.spec.ts

```diff
@@ -9,54 +9,27 @@ const TEST_DATA_SOURCES = {
   },
 };
 
-describe('ApiClient', () => {
-  const apiBaseUrlCases = [
-    ['https://localhost:3000/', 'https://localhost:3000/api/datasources/proxy/uid/grafanacloud-profiles-test'],
-    [
-      'https://firedev001.grafana-dev.net/',
-      'https://firedev001.grafana-dev.net/api/datasources/proxy/uid/grafanacloud-profiles-test',
-    ],
-    // app URL with pathname
-    [
-      'https://admin-dev-us-central-0.grafana-dev.net/stable-grafana/',
-      'https://admin-dev-us-central-0.grafana-dev.net/stable-grafana/api/datasources/proxy/uid/grafanacloud-profiles-test',
-    ],
-    // app URL with no slash at the end
-    [
-      'https://admin-dev-us-central-0.grafana-dev.net/stable-grafana',
-      'https://admin-dev-us-central-0.grafana-dev.net/stable-grafana/api/datasources/proxy/uid/grafanacloud-profiles-test',
-    ],
-  ];
-
-  describe.each(apiBaseUrlCases)('when the app URL provided by the platform is "%s"', (appUrl, expectedApiBaseUrl) => {
-    test(`the API base URL is "${expectedApiBaseUrl}"`, () => {
-      // testing like it's 2023 :man_shrug:
-      jest.doMock('@grafana/runtime', () => ({
-        config: {
-          appUrl,
-          datasources: TEST_DATA_SOURCES,
-        },
-      }));
-
-      const { ApiClient } = require('../ApiClient');
-
-      const apiClient = new ApiClient();
-
-      expect(apiClient.baseUrl).toBe(expectedApiBaseUrl);
-    });
-  });
+function setupMocks(options?: { appUrl?: string; bootData?: Record<string, any>; dataSources?: Record<string, any> }) {
```

Comment:
I took the opportunity to refactor these tests by creating the `setupMocks` helper function.